packages/components/avatar/src/avatar.ts

```diff
@@ -1,85 +1,36 @@
-import { type AvatarConfig, Config } from '@sl-design-system/shared';
+import { Badge } from '@sl-design-system/badge';
 import { Tooltip } from '@sl-design-system/tooltip';
-import { type CSSResultGroup, LitElement, type PropertyValues, type TemplateResult, html, nothing, svg } from 'lit';
+import { type CSSResultGroup, LitElement, type PropertyValues, type TemplateResult, html, nothing } from 'lit';
 import { property, state } from 'lit/decorators.js';
+import { ifDefined } from 'lit/directives/if-defined.js';
+import { styleMap } from 'lit/directives/style-map.js';
 import styles from './avatar.scss.js';
-import {
-  type AvatarBadge,
-  type AvatarFallbackType,
-  type AvatarIcon,
-  type AvatarImage,
-  type AvatarSize,
-  type UserStatus
-} from './models.js';
 
 declare global {
   interface HTMLElementTagNameMap {
     'sl-avatar': Avatar;
   }
 }
 
-const BORDER_WIDTH = 4; //has to be double the desired "gap"; the stroke is centered on the path, so only half of is it outside the badge rect.
-const FOCUS_RING_STROKE_WIDTH = 2;
-const FOCUS_RING_STROKE_OFFSET = 2;
-
-const IMAGE_SIZES: Record<AvatarSize, number> = {
-  sm: 24,
-  md: 32,
-  lg: 40,
-  xl: 52,
-  '2xl': 64,
-  '3xl': 80
-};
-
-const BADGE_SIZES: Record<AvatarSize, number> = {
-  sm: 8,
-  md: 12,
-  lg: 14,
-  xl: 16,
-  '2xl': 18,
-  '3xl': 20
-};
-
-const ICON_SIZES: Record<AvatarSize, number> = {
-  sm: 14,
-  md: 16,
-  lg: 18,
-  xl: 24,
-  '2xl': 32,
-  '3xl': 32
-};
-
-// offset relative to image; same principle as with css-positioning.
-const OFFSET_CIRCLE: Record<AvatarSize, number> = {
-  sm: -2,
-  md: -4,
-  lg: -4,
-  xl: -2,
-  '2xl': -2,
-  '3xl': 2
-};
-
-const OFFSET_SQUARE: Record<AvatarSize, number> = {
-  sm: -4,
-  md: -6,
-  lg: -7,
-  xl: -8,
-  '2xl': -9,
-  '3xl': -10
-};
+export type AvatarSize = 'sm' | 'md' | 'lg' | 'xl' | '2xl' | '3xl';
 
 /**
  * An avatar component to show a picture, initials or icon, to provide a quickly recognizable representation of a user.
  *
  * ```html
- *   <sl-avatar display-name="Lynn Smith" picture="http://sanomalearning.design/avatars/lynn.png"></sl-avatar>
+ *   <sl-avatar display-name="Lynn Smith" picture-url="http://sanomalearning.design/avatars/lynn.png"></sl-avatar>
  * ```
  *
+ * @csspart avatar - The container for positioning the badge.
+ * @csspart initials - The initials to display when no picture is set.
  * @csspart name - The display name, either a <span> or <a> if `href` is set.
+ * @csspart picture - The element containing the image, initials or fallback content.
  * @csspart wrapper - The wrapper element around the image and name.
  *
- * @cssprop --sl-avatar-max-inline-size - Max inline-size of the container in vertical mode. If not set it will behave like a regular `display: block` element.
+ * @slot badge - The badge to display on the avatar.
+ * @slot default - The subheading of the avatar.
+ * @slot fallback - The fallback content to display when no picture is set.
```

Comment:
Why would you want this as a slot? You want this to be the same over your entire application (with a possible overwrite, but not sure why you would want that.) so I would suggest an `sl-icon` which we choose.

Reply:
For the sl-icon we can use the same setup we have for the breadcrumbs; there we use an icon named `breadcrumb-separator`, that points to the `slash-forward` icon by default but for magister it's `chevron-right`